### DISCUSSION TITLE
[1104] 회의록 - 모델 분배

### DISCUSSION THREAD

**Discussion by SooMiiii** [boostcampaitech7/level2-bookratingprediction-recsys-02#7] [Daily Report]
### 다룬 내용
- EDA & 전처리

베이스라인 전처리 확인
context 거의 결측치 처리 -> 최빈값
>나이 -> 다른 방식으로 대체할 수 있을까?
주소 na 값 많음 
주소 이상치 -> 처리중
city, country가 같으면 state 통일 시키기

지피티에서 아이디어 얻기

책별로 평점 수
출판사 카테고리화

- wandb 팀으로 되는듯! 활용하자

### 역할 분배
모델을 나누자!
![image](https://github.com/user-attachments/assets/903f2a38-645c-41b4-9bfc-94756de44a45)
차례대로 수미-현구-도걸-윤혜-동준
시간 남으면 boosting model 도 적용

**Comment by seoo2001** [comment]
## 모델링 아이디어
리뷰가 1개인 유저에 대해서는 User-Item 데이터를 활용하지 말아보기. 1개의 리뷰가 유저-아이템 관계를 설명하기 어려울 것으로 가정. -> 컨텐츠 기반 모델 적용
- 유저 별 리뷰 개수
  - 1개인 유저 수: 34617
  - 2개 이상인 유저 수: 25186
- 리뷰 1개 쓴 유저가 등장하는 test row 개수: 3860 (5%) 
  - 5%의 데이터에만 다른 모델 적용

**Comment by doffice0827** [comment]
## GPT 추천

1. 사용자 관련 feature
사용자 평균 평점 (user_avg_rating): train_ratings.csv에서 특정 사용자가 부여한 모든 평점의 평균값. 사용자의 평가 성향을 반영할 수 있습니다.
사용자의 평점 분포 (user_rating_std): 사용자가 부여한 평점의 표준 편차. 평점을 일관되게 부여하는지, 가변성이 큰지를 알 수 있습니다.
연령대 (age_group): users.csv의 age를 나이대 그룹으로 묶어 추가할 수 있습니다. 예를 들어, 0–18, 19–30, 31–45, 46–60, 61+ 등으로 나눌 수 있습니다.
지역 기반 특성 (location_country): users.csv의 location에서 국가명을 추출하여 사용자가 거주하는 국가를 feature로 추가하면 국가별 차이 등을 학습할 수 있습니다.
2. 책 관련 feature
책의 평균 평점 (book_avg_rating): train_ratings.csv에서 특정 책에 대해 부여된 모든 평점의 평균값. 이 책이 다른 사용자들에게 어떤 평가를 받았는지 알 수 있습니다.
평가된 횟수 (book_rating_count): 특정 책에 대해 평점을 남긴 사용자 수. 많이 평가된 책이 인기 있거나 평점이 더 안정적일 가능성이 높습니다.
발행 연도 변형 (book_age): books.csv의 year_of_publication을 사용해 책이 얼마나 오래된 것인지 계산하여 feature로 추가합니다.
저자 기반 특성 (author_avg_rating): 동일 저자의 다른 책에 대한 평점 평균값. 저자에 대한 선호도나 인지도가 반영될 수 있습니다.
3. 사용자-책 상호작용 feature
사용자-카테고리 상관성 (user_category_preference): 특정 사용자가 자주 평점을 준 책의 category를 집계하여, 사용자가 선호하는 카테고리를 feature로 추가할 수 있습니다.
사용자-출판사 상관성 (user_publisher_preference): 특정 사용자가 주로 평가한 publisher를 기록하여 출판사에 대한 선호도 정보를 feature로 추가할 수 있습니다.

**Comment by doffice0827** [comment]
WDN DCN NCF 다양한 학습 결과

------ WDN ------
baseline + basic_data 37-38 수정
saved/submit/20241105_142920_WDN.csv
wandb: Train RMSE 1.32925
wandb: Valid MAE 1.71648
wandb: Valid MSE 5.52551
wandb: Valid RMSE 2.34986

baseline + epochs 30
saved/submit/20241105_144932_WDN.csv
wandb: Train RMSE 1.27889
wandb: Valid MAE 1.71851
wandb: Valid MSE 5.55953
wandb: Valid RMSE 2.3571

baseline + gpt 추천
saved/submit/20241105_145938_WDN.csv
wandb: Train RMSE 1.43078
wandb: Valid MAE 1.69648
wandb: Valid MSE 5.45799
wandb: Valid RMSE 2.33542

baseline + 복잡도 낮춰서
saved/submit\20241105_161155_WDN.csv
wandb: Train RMSE 1.44986
wandb: Valid MAE 1.70102
wandb: Valid MSE 5.23763
wandb: Valid RMSE 2.28803

baseline + 복잡도 낮춰서 + epoch 4
saved/submit\20241105_162407_WDN.csv
wandb: Train RMSE 1.71636
wandb: Valid MAE 1.69922
wandb: Valid MSE 4.98407
wandb: Valid RMSE 2.23191


------ DCN ------
baseline + basic_data 37-38 수정
saved/submit/20241105_150616_DCN.csv
wandb: Train RMSE 1.58049
wandb: Valid MAE 1.91578
wandb: Valid MSE 6.06117
wandb: Valid RMSE 2.46128

baseline + epochs 30
saved/submit/20241105_151313_DCN.csv
wandb: Train RMSE 1.50944
wandb: Valid MAE 1.89184
wandb: Valid MSE 6.21423
wandb: Valid RMSE 2.49212

baseline + cross_layer_num 4
saved/submit/20241105_152518_DCN.csv
wandb: Train RMSE 1.52941
wandb: Valid MAE 1.8252
wandb: Valid MSE 5.75286
wandb: Valid RMSE 2.3978

baseline + cross_layer_num 5
saved/submit/20241105_153213_DCN.csv
wandb: Train RMSE 1.54501
wandb: Valid MAE 1.75313
wandb: Valid MSE 5.63947
wandb: Valid RMSE 2.37429

baseline + gpt 추천
saved/submit/20241105_153820_DCN.csv
wandb: Train RMSE 1.69925
wandb: Valid MAE 1.72113
wandb: Valid MSE 5.58712
wandb: Valid RMSE 2.36303


------ NCF ------
baseline + basic_data 37-38 수정
saved/submit/20241105_154607_NCF.csv
wandb: Train RMSE 1.42588
wandb: Valid MAE 1.8074
wandb: Valid MSE 5.62682
wandb: Valid RMSE 2.37147

baseline + epochs 30
saved/submit\20241105_155145_NCF.csv
wandb: Train RMSE 1.24697
wandb: Valid MAE 1.81364
wandb: Valid MSE 5.74475
wandb: Valid RMSE 2.39617

baseline + gpt 추천
saved/submit/20241105_155904_NCF.csv
wandb: Train RMSE 1.42219
wandb: Valid MAE 1.71872
wandb: Valid MSE 5.57066
wandb: Valid RMSE 2.35943
